Project1/Map.cpp

```diff
@@ -123,14 +124,13 @@ PathMap Map::findShortestPaths(GridCell* start, int max_move, MobilityList<bool>
 			std::optional<int> cost = edge.getCost(mobility, intangible);
 			if (cost.has_value()) {
 				cost = top.first + cost.value();
-				if (cost.value() <= max_move && (path_map.count(edge._cell) == 0 || cost.value() < path_map[edge._cell].first)) {
-					path_map.insert_or_assign(edge._cell, std::pair<int, GridCell*>(cost.value(), top.second));
-					queue.emplace(cost.value(), edge._cell);
+				if (cost.value() <= max_move && (path_map.count(&(edge._cell)) == 0 || cost.value() < path_map[&edge._cell].first)) {
+					path_map.insert_or_assign(&edge._cell, std::pair<int, GridCell*>(cost.value(), top.second));
+					queue.emplace(cost.value(), &edge._cell);
```

Comment:
You can make your map and queue contain GridCell& so you don't have to keep using &